src/fondant/component/executor.py

```diff
@@ -569,12 +549,16 @@ def _execute_component(
 
         # Clear divisions if component spec indicates that the index is changed
         if self._infer_index_change():
+            # TODO: might causing issues for merging components
```

Comment:
We still need divisions, but we need to make sure to reset and recalculate them if the index changes. See the [Dask documentation](https://docs.dask.org/en/stable/generated/dask.dataframe.DataFrame.map_partitions.html#:~:text=Also%20note%20that%20the%20index%20and%20divisions%20are%20assumed%20to%20remain%20unchanged.%20If%20the%20function%20you%E2%80%99re%20mapping%20changes%20the%20index/divisions%2C%20you%E2%80%99ll%20need%20to%20clear%20them%20afterwards%3A).